tests/foreman/api/test_repository.py

```diff
@@ -1738,81 +1767,49 @@ class SRPMRepositoryTestCase(APITestCase):
 
     @classmethod
-    @skip_if_bug_open('bugzilla', 1378442)
     def setUpClass(cls):
         """Create a product and an org which can be re-used in tests."""
         super(SRPMRepositoryTestCase, cls).setUpClass()
         cls.org = entities.Organization().create()
         cls.product = entities.Product(organization=cls.org).create()
+        cls.lce = entities.LifecycleEnvironment(organization=cls.org).create()
 
     @tier2
-    def test_positive_sync(self):
-        """Synchronize repository with SRPMs
-
-        :id: f87391c6-c18a-4c4f-81db-decbba7f1856
-
-        :expectedresults: srpms can be listed in repository
-        """
-        repo = entities.Repository(
-            product=self.product,
-            url=FAKE_YUM_SRPM_REPO,
-        ).create()
-        repo.sync()
-        result = ssh.command(
-            'ls /var/lib/pulp/published/yum/https/repos/{}/Library'
-            '/custom/{}/{}/ | grep .src.rpm'
-            .format(
-                self.org.label,
-                self.product.label,
-                repo.label,
-            )
-        )
-        self.assertEqual(result.return_code, 0)
-        self.assertGreaterEqual(len(result.stdout), 1)
-
-    @tier2
-    def test_positive_sync_publish_cv(self):
+    def test_positive_repo_sync_publish_promote_cv(self):
         """Synchronize repository with SRPMs, add repository to content view
-        and publish content view
+        and publish, promote content view
 
-        :id: a0868429-584c-4e36-b93f-c85e8e94a60b
+        :id: f87391c6-c18a-4d4f-81db-decbba7f1856
 
-        :expectedresults: srpms can be listed in content view
+        :expectedresults: srpms can be listed in organization, content view, Lifecycle env
         """
-        repo = entities.Repository(
-            product=self.product,
-            url=FAKE_YUM_SRPM_REPO,
-        ).create()
-        repo.sync()
+        repo = entities.Repository(product=self.product).create()
+        entities.ContentUpload(repository=repo).upload(filepath=get_data_file(SRPM_TO_UPLOAD),
+                                                       content_type='srpm')
         cv = entities.ContentView(organization=self.org).create()
         cv.repository = [repo]
         cv.update(['repository'])
         cv.publish()
-        result = ssh.command(
-            'ls /var/lib/pulp/published/yum/https/repos/{}/content_views/{}'
-            '/1.0/custom/{}/{}/ | grep .src.rpm'
-            .format(
-                self.org.label,
-                cv.label,
-                self.product.label,
-                repo.label,
-            )
-        )
-        self.assertEqual(result.return_code, 0)
-        self.assertGreaterEqual(len(result.stdout), 1)
+        self.assertEqual(cv.repository[0].read().content_counts['srpm'], 3)
+        self.assertEqual(len(entities.Srpms().search(query={'organization_id': self.org.id})), 3)
+
+        cv = cv.read()
```

Comment:
Shouldn't that be called after `cv.publish()`, before assertions?

Also, usually we re-assign new data back with update:
```python
cv = cv.update(['repository'])
```
So maybe drop this line completely. I think publishing doesn't change anything in CV object.

Reply:
1) It made intentionally to call it if below assertion pass else not require to call further and test would be fail:
```
self.assertEqual(cv.repository[0].read().content_counts['srpm'], 3)
self.assertEqual(len(entities.Srpms().search(query={'organization_id': self.org.id})), 3)
``` 
And same way promote is also done (to save test time and made assertion more aligned with action only instead of wait at the end)

2)  the end goal to read is (cv.version) and that only getting with `cv.read()` ... I tried below way but it only comes with same:
 
```
(Pdb) cv = cv.update(['repository'])
(Pdb) cv.publish()
(Pdb) cv.version
[]
(Pdb) cv = cv.read()
(Pdb) cv.version
[nailgun.entities.ContentViewVersion(id=21)]
(Pdb) 
```

Reply:
Ad 2: ah I see. Agreed.
Ad 1: For me, it felt unusual that you update the object, do not assign it back to variable and run assertions on old version, not newest one you received from server. But more I think about it, more I realize improvements I could propose are superficial.

Let's leave it as it is. Thanks for explanations.